drivers/bh1900nux/bh1900nux.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2020 Nalys
+ *  
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_bh1900nux
+ * @{
+ *
+ * @file
+ * @brief       BH1900NUX temperature sensor driver implementation
+ *
+ * @author      Wouter Symons <wsymons@nalys-group.com> 
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "bh1900nux.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+int bh1900nux_init(bh1900nux_t *dev, const bh1900nux_params_t *params)
+{
+    if(dev == NULL | params == NULL) {
+        return -1;
+    }
+
+    /* initialize the device descriptor */
+    dev->i2c = params->i2c;
+    dev->addr = params->addr;
+
+    return BH1900NUX_OK;
+}
+
+int bh1900nux_read(const bh1900nux_t *dev, double *temp)
+{
+    int ret = 0;
+    int tmp = 0;
+    uint8_t bytes[2] = {0};
+
+    /* Read raw sensor value */
+    DEBUG("[bh1900nux] read temperature\n");
+    ret = i2c_acquire(dev->i2c);
+    if(ret < 0) {
+        return BH1900NUX_ERR_I2C;
+    }
+    ret = i2c_read_regs(dev->i2c, dev->addr, BH1900NUX_REG_ADDR, bytes, 2, 0);
+    if(ret < 0) {
+        return ret;
+    }
+    i2c_release(dev->i2c);
+
+    /* Calculate temperature */
+    tmp = ((uint16_t)bytes[0] << 8) | bytes[1];
+    *temp = 0.0625 * ((double) tmp / 16);
```

Comment:
Shouldn't dividing by 16 once be enough? 😉 
You can avoid the floating point arithmetic entirely by multiplying by 1000 first and then divide by 16, returning mC.

Reply:
Now I get the data format.
Forget about the `ntohs()`:

`bytes[0]` has the whole °C - you can use this as is, or multiply it by 1000 for mC.
`bytes[1]` has the fractional part * 16 - just shift right by 4 places to get the fractional part in mC.

Reply:
Resolved the first comment, second part is still up for debate (discussed on IRC).